server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java

```diff
@@ -722,6 +726,20 @@ private Collection<DiscoveryNode> getMasterEligibleNodes() {
         return masterEligibleNodes;
     }
 
+    /**
+     * Returns a random master eligible node, or null if this node does not know about any master eligible nodes
+     * @return A random master eligible node or null
+     */
+    // Non-private for unit testing
+    @Nullable
+    DiscoveryNode getRandomMasterEligibleNode() {
+        Collection<DiscoveryNode> masterEligibleNodes = getMasterEligibleNodes();
```

Comment:
We could probably just make this a `Set<DiscoveryNode>`, since I don't think we need it to be generic for a reason right? Then we could avoid converting it into an array just to randomly pick one of the elements.

(Not that we have to do that here, maybe an idea for the future)

Reply:
Yeah I couldn't remember why I'd made it a Collection (I assume just to make it no more specific than it had to be at the time?). But I think even if it was a Set we'd have to do something like this (or something different but no better really).